cli/resource-mover-cli/src/main/kotlin/net/thunderbird/cli/resource/mover/StringResourceMover.kt

```diff
@@ -0,0 +1,186 @@
+        println("    Moving key to file: ${targetFile.path}")
+
+        if (containsKey(targetFile, key)) {
+            println("    Key already exists in target file: ${targetFile.path} replacing it.")
+            replaceKeyInTarget(targetFile, keyDeclaration, key)
+        } else {
+            addKeyToTarget(targetFile, keyDeclaration)
+        }
+    }
+
+    private fun addKeyToTarget(targetFile: File, keyDeclaration: String) {
+        val targetContent = StringBuilder()
+
+        targetFile.forEachLine { line ->
+            if (line.contains(RESOURCE_CLOSING_TAG)) {
+                targetContent.appendLine(keyDeclaration.trimEnd())
+                targetContent.appendLine(line)
+            } else {
+                targetContent.appendLine(line)
+            }
+        }
+
+        targetFile.writeText(targetContent.toString())
+    }
+
+    private fun replaceKeyInTarget(targetFile: File, keyDeclaration: String, key: String) {
+        println("    Replacing key in file: ${targetFile.path}")
+
+        val oldKeyDeclaration = extractKeyDeclaration(targetFile, key)
+        val targetContent = targetFile.readText()
+
+        targetFile.writeText(targetContent.replace(oldKeyDeclaration, keyDeclaration))
+    }
+
+    private fun deleteKeyFromSource(sourceFile: File, keyDeclaration: String) {
+        println("    Deleting key from file: ${sourceFile.path}")
+
+        val sourceContent = sourceFile.readText()
+
+        sourceFile.writeText(sourceContent.replace(keyDeclaration, ""))
+    }
+
+    private fun isSourceFileEmpty(sourceFile: File): Boolean {
+        val sourceContent = sourceFile.readText()
+        return sourceContent.contains(STRING_CLOSING_TAG).not() && sourceContent.contains(PLURALS_CLOSING_TAG).not()
+    }
+
+    private fun openOrCreateTargetFile(targetPath: File, sourceFile: File): File {
+        val targetFilePath = targetPath.resolve(sourceFile.parentFile.name)
+        val targetFile = File(targetFilePath, sourceFile.name)
+        val targetDirectory = targetFile.parentFile
+
+        if (!targetDirectory.exists()) {
+            targetDirectory.mkdirs()
+            println("    Target directory created: ${targetDirectory.path}")
+        }
+
+        if (!targetFile.exists()) {
+            createTargetFile(targetFile)
+        }
+
+        return targetFile
+    }
+
+    private fun createTargetFile(targetFile: File) {
+        val isNewFileCreated: Boolean = targetFile.createNewFile()
+        if (!isNewFileCreated) {
+            printError("Target file could not be created: ${targetFile.path}")
+            exitProcess(-1)
+        }
+
+        targetFile.writeText(TARGET_FILE_CONTENT)
+        println("Target file ${targetFile.path} created")
+    }
+
+    private fun printError(message: String) {
+        System.err.println("\n$message\n")
+    }
+
+    private companion object {
+        const val RESOURCE_PATH = "/src/main/res/"
```

Comment:
I understand your concern about hardcoded path separators, but given our core team uses Unix-based systems, I think it's reasonable to prioritize compatibility with those platforms. This refactoring tool is intended for internal use and not designed for external contributions. 

If we want to enable contributions from Windows systems, we should consider validating the project's compatibility on that platform first. This would require having a Windows system available, which could be addressed in a separate issue with discussion on the necessary resources and effort required.